server/src/main/java/org/cloudfoundry/identity/uaa/scim/jdbc/JdbcScimUserProvisioning.java

```diff
@@ -89,6 +91,10 @@ public Log getLogger() {
 
     public static final String CHANGE_PASSWORD_SQL = "update users set lastModified=?, password=?, passwd_lastmodified=? where id=? and identity_zone_id=?";
 
+    public static final String ADD_PASSWORD_HISTORY_SQL = "insert into password_history (user_id, identity_zone_id, password, changed) values (?,?,?,?)";
+
+    public static final String GET_PASSWORD_HISTORY_SQL = "select password from password_history where user_id=? and identity_zone_id=? order by changed desc limit ?";
+
```

Comment:
These are the 2 new queries - adding the password to the history table and fetching passwords for a particular user in temporal order.